### ISSUE TITLE
Question about training iterations and epoch

### ISSUE THREAD

**Issue by SmilingFrank** [jakeret/tf_unet#76]
It's more of a question than an issue, I think that I am misunderstanding the parameters.

Is the training iteration parameter the number of batch used for each epoch? I was slightly confused at first because I expected the epoch to be how many time it went through my whole dataset.

So, if I have a dataset of 4000 images and 4000 masks and I want to go through my whole dataset 10 times, should I set the training iteration parameter to 4000 and epoch to 10?

**Comment by jakeret** [comment]
`training_iters` is controlling the size of your mini batch.
`epochs` defines how many mini batches should be executed. After each epoch the learning rate is adapted.

As for your particular case, this is something I would experiment with. However, 10 epochs seems to be rather small.
I typically use something like `training_iters=32` and `epochs=100`

**Comment by SmilingFrank** [comment]
Ah, I see. Thank you for your answer. So, if I understand correctly, the parameters training_iters=32 and epochs=100 should work for a dataset of a few thousands examples.
